include/wallet_manager.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include "context.h"
+#include "task.h"
+#include "thread_pool/strand.hpp"
+
+#include <atomic>
+#include <string>
+#include <memory>
+#include <vector>
+
+namespace tools
+{
+
+class GraftWallet;
+
+}
+
+namespace graft
+{
+
+/// Wallet manager
+class WalletManager
+{
+public:
+    using WalletId = std::string;
+    using Url = std::string;
+
+    struct TransferDestination
+    {
+      std::string address;
+      uint64_t    amount;
+
+      TransferDestination(const std::string& in_address, uint64_t in_amount)
+        : address(in_address), amount(in_amount) {}
+    };
+
+    using TransferDestinationArray = std::vector<TransferDestination>;
+
+    // Constructors / destructor
+    WalletManager(TaskManager& task_manager, bool testnet = false);
+    ~WalletManager();
+    WalletManager(const WalletManager&) = delete;
+    WalletManager& operator = (const WalletManager&) = delete;
+
+    /// Create account
+    void createAccount(Context&, const std::string& password, const std::string& language, const Url& callback_url = Url());
+
+    /// Restore account
+    void restoreAccount(Context&, const std::string& password, const std::string& seed, const Url& callback_url = Url());
+
+    /// Request balance
+    void requestBalance(Context&, const WalletId&, const std::string& account_data, const std::string& password, const Url& callback_url = Url());
+
+    /// Prepare transfer
+    void prepareTransfer(Context&, const WalletId&, const std::string& account_data, const std::string& password, const TransferDestinationArray& destinations, const Url& callback_url = Url());
+
+    /// Flush disk caches
+    void flushDiskCaches();
+
+private:
+    struct WalletHolder;
+    using WalletPtr = std::shared_ptr<WalletHolder>;
+
+private:
```

Comment:
I don't know why, but we have a style to have single public, private and protected sections per class (if possible). So private here excessive.

Reply:
Fixed

Reply:
Fixed